### PR TITLE
add storm compatibility docs

### PR DESCRIPTION
**Pull request by manuzhang**
[Review on Reviewable](https://reviewable.io/reviews/gearpump/gearpump/1525)

### CODE REVIEW

File: docs/dev-storm.md [hunk 1]
```diff
@@ -1,6 +1,80 @@
...
+
+At runtime, spouts and bolts are running inside `StormProducer` tasks and `StormProcessor` tasks respectively. Messages emitted by spout are passed to `StormProducer`, transferred to `StormProcessor` and passed down to bolt.  Messages are serialized / deserialized with Storm serializers.
+
+Storm ackers are dropped since Gearpump has a different mechanism of message tracking and flow control. 
+
+### Message tracking 
+
+Storm tracks the lineage of each message with ackers to guarantee at-least-once message delivery. Failed messages are re-sent from spout.
+
+Gearpump [tracks messages between a sender and receiver in a efficient way](gearpump-internals.html#how-do-we-detect-message-loss). Message loss causes the whole application to replay from the [global minimum clock](gearpump-internals.html#application-clock-and-global-clock-service). This doesn't provide at-least-once delivery (read more about at-least-once in Gearpump [here](gearpump-internals.html#at-least-once-message-delivery-and-kafka)). That is to say, Storm applications run with at-most-once on Gearpump unless the application itself ensures at-least-once or exactly-once.
```
Comment (clockfly) [inline docs/dev-storm.md]: ···
This doesn't provide at-least-once delivery
···

At least once no longer supported? why?

File: docs/dev-storm.md [hunk 2]
```diff
@@ -1,6 +1,80 @@
...
+  
+   ```
+   ./target/pack/bin/local
+   ```
+
+2. submit a topology from storm-starter
+
+   ```
+   bin/storm -verbose -config storm.yaml -jar storm-starter-${STORM_VERSION}.jar storm.starter.ExclamationTopology exclamation 
+   ```
```
Comment (clockfly) [inline docs/dev-storm.md]: Maybe document that submission via UI is not supported yet.

File: docs/dev-storm.md [hunk 3]
```diff
@@ -1,6 +1,80 @@
...
+Storm throttles flow rate at spout, which stops sending messages if the number of unacked messages exceeds `topology.max.spout.pending`. 
+
+Gearpump has flow control between tasks such that [sender cannot flood receiver](gearpump-internals.html#how-do-we-do-flow-control), which is backpressured till the source.
+
+### Configurations
+
+All Storm configurations are respected with the following priority order 
+
+```
+defaults.yaml < storm.yaml < application config < component config < config file set with "-config"
```
Comment (clockfly) [inline docs/dev-storm.md]: What is 

 application config?

 < component config?

 < config file set with "-config"?

File: docs/dev-storm.md (hunk 1 above)
Comment (manuzhang) [inline docs/dev-storm.md]: Acker bolts are added by storm system and bring considerate message overhead.  I think we already have our ways to do flow control and message tracking.

Comment (clockfly) [inline docs/dev-storm.md]: I don't get your point.

So we support at least once. right? Why you say,
```
This doesn't provide at-least-once delivery
```

Comment (manuzhang) [inline docs/dev-storm.md]: We don't support at-least-once for storm application yet. Our at-least-once support requires a `TimeReplayableSource`. Replaying from global minimum clock doesn't mean we resend the failed messages.

**Comment by clockfly** [comment]
It is not acceptable to sacrifice feature like at-least-once. At least for well known connector like kafka.

**Comment by manuzhang** [comment]
At-least-once is not dropped for Kafka since its offset is stored in Zookeeper. On restart, Storm's `KafkaSpout` will be able to continue from the last committed offset. At-least-once holds for all other Storm connectors where message or message offset is managed by Storm application itself.

**Comment by clockfly** [comment]
Will the kafka offset in zookeeper be moved forward? how?

In Storm runtime, Kafka spout will commit offset when there is ack success. How is this handled in current gearpump runtime?

**Comment by clockfly** [comment]
link with https://github.com/gearpump/gearpump/issues/1536
Move further discussion there. 

+1 on current doc.

**Pull request merged**